src/Data/Bin/Properties.agda

```diff
@@ -1,143 +1,904 @@
 ------------------------------------------------------------------------
 -- The Agda standard library
 --
--- Properties of the binary representation of natural numbers
+-- Arithmetic properties related to addition and multiplication of
+-- binary represented natural numbers.
 ------------------------------------------------------------------------
 
 {-# OPTIONS --without-K --safe #-}
 
 module Data.Bin.Properties where
 
-open import Data.Bin
-open import Data.Digit using (Bit; Expansion)
-import Data.Fin as Fin
-import Data.Fin.Properties as 𝔽ₚ
-open import Data.List.Base using (List; []; _∷_)
-open import Data.List.Properties using (∷-injective)
-open import Data.Nat
-  using (ℕ; zero; z≤n; s≤s)
-  renaming (suc to 1+_; _+_ to _+ℕ_; _*_ to _*ℕ_; _≤_ to _≤ℕ_)
-import Data.Nat.Properties as ℕₚ
-open import Data.Product using (proj₁; proj₂)
-open import Function using (_∘_)
-open import Relation.Binary
-open import Relation.Binary.Consequences
-open import Relation.Binary.PropositionalEquality
-  using (_≡_; _≢_; refl; sym; isEquivalence; resp₂; decSetoid)
-open import Relation.Nullary using (yes; no)
+open import Algebra using (Magma; Semigroup; Monoid; CommutativeMonoid;
+                                             Semiring; CommutativeSemiring)
+import      Algebra.FunctionProperties as FuncProp
+open import Algebra.Structures using
+            (IsMagma; IsSemigroup; IsMonoid; IsCommutativeMonoid; IsSemiring;
+             IsSemiringWithoutAnnihilatingZero; IsCommutativeSemiring
+            )
+open import Data.Bin.Base using (Bin; _≟_; suc; pred; _+_; _*_; 2*; 2^; size;
+                                                                1#; 2#; toℕ)
+open import Data.Nat using (ℕ; s≤s)
```

Comment:
As previously discussed, instead of renaming everything we typically use `open import Data.Nat as ℕ using (ℕ; s≤s)` and then write `ℕ.suc`, `_ℕ.+_` etc.

Reply:
I use this for  Data.Nat.Properties  as ℕp.  For example, ℕp.+-assoc.

But for  Data.Nat as ``ℕ``,  this occurs not nice, for the following reasons.
1) The sections like ``(_ℕ.+ x),  (x ℕ.+_)``  (used in proof combinators) are not type-checked.
2) The expression
```  
                          (x +ₙ y)  *ₙ  (x +ₙ z)
```
is easier to read than
```
                          (x ℕ.+ y)  ℕ.*  (x ℕ.+ z),
```
the lower indice ``ₙ`` is a mnemonic for ``ℕ``.
And there are (and are expected) many similar expressions.

Do you agree with my version?